analyses/cnv-chrom-plot/util/bin-coverage.R

```diff
@@ -0,0 +1,143 @@
+  # Given a sample_id, CN segment ranges, and binned genome ranges object, 
+  # make a call for each bin on what CN copy status has the most coverage in the bin. 
+  # Uses bp_per_bin function. 
+  #
+  # Args:
+  #   sample_id: A string that corresponds to the biospecimen s
+  #   seg_ranges: A GenomicRanges object that contains a `status` and a `biospecimen` slot.  
+  #               The `biospecimen slot will be used to split out the `sample_id`'s corresponding ranges.   
+  #               The `status` slot should have the that has the gain/loss/neutral 
+  #               status call for each segment. Data will be split up by this variable. 
+  #   bin_ranges: A binned GenomicRanges made from tileGenome that has been uncompressed with `unlist`. 
+  #   perc_delta_threshold: What covereage percent difference do you need to make the call?
+  #
+  # Returns:
+  #  a small data.frame that contains the status call of the sample for each bin. 
+  #
+  # Extract the ranges for this sample
+  sample_seg_ranges <- seg_ranges[which(seg_ranges$biospecimen == sample_id)]
+  
+  # Split ranges into their respective statuses
+  gain_ranges <- sample_seg_ranges[sample_seg_ranges$status == "gain"]
+  loss_ranges <- sample_seg_ranges[sample_seg_ranges$status == "loss"]
+  neutral_ranges <- sample_seg_ranges[sample_seg_ranges$status == "neutral"]
+  
+  # Calculate length of each type of status per bin
+  gain_per_bin <- bp_per_bin(bin_ranges, gain_ranges)
+  loss_per_bin <- bp_per_bin(bin_ranges, loss_ranges)
+  neutral_per_bin <- bp_per_bin(bin_ranges, neutral_ranges)
+  
+  # Format this data into one data.frame where each row is a bin
+  bin_bp_status <- data.frame(
+    bin = as.numeric(1:length(bin_ranges)),
+    # Keep bin width
+    bin_width = bin_ranges@ranges@width
+  ) %>%
+    # Join gains coverage data
+    dplyr::left_join(gain_per_bin,
+                     by = "bin"
+    ) %>%
+    # Join loss coverage data
+    dplyr::left_join(loss_per_bin,
+                     by = "bin",
+                     suffix = c(".gain", ".loss")
+    ) %>%
+    # Join neutral coverage data
+    dplyr::left_join(neutral_per_bin,
+                     by = "bin"
+    ) %>%
+    # If there is an NA, at this point we can assume it means 0
+    dplyr::mutate_at(
+      dplyr::vars(
+        dplyr::starts_with("bp_per_bin")
+      ),
+      ~ tidyr::replace_na(., 0)
+    ) %>%
+    # Reformat neutral so it is like the others
+    dplyr::rename(bp_per_bin.neutral = bp_per_bin) %>%
+    # Calculate the bins percentage of each status
+    dplyr::mutate(
+      perc_gain = bp_per_bin.gain / bin_width,
+      perc_loss = bp_per_bin.loss / bin_width,
+      perc_neutral = bp_per_bin.neutral / bin_width
+    ) %>%
+    # Use these percentages for declaring final call per bin based on
+    # the perc_delta_threshold
+    dplyr::mutate(
+      status = dplyr::case_when(
+        (perc_gain - perc_loss) > perc_delta_threshold ~ "gain",
```

Comment:
Would you further explain the logic here to me? Perhaps update the comment relevant to this step as well.

If we want to accurately call a gain by subtracting the non-gain percentage calls then I believe we would also want to include the `perc_neutral` in this logic. 

Also, it appears that you are calculating `perc_neutral` here but it is not being used. So if we decide to keep it out of the logic here then we can remove it but let's establish what we want here first.